### ISSUE TITLE
Potentially incorrect aggregation on node_namespace_pod_container:container_memory_working_set_bytes

### ISSUE THREAD

**Issue by AndresCalimero** [kubernetes-monitoring/kubernetes-mixin#813]
Hi, when I run the recorded query `node_namespace_pod_container:container_memory_working_set_bytes{container="neo4j"}` I get the following results:
![Screenshot from 2022-12-14 13-40-48](https://user-images.githubusercontent.com/13048718/207610792-9494521c-b37b-4f16-8531-49321e4e00de.png)
I've two k8s clusters (staging and production), each one is running a Neo4j database cluster, and as you can see, the nodes are duplicated, those nodes exist only in the production cluster, when I run the query `container_memory_working_set_bytes{image!="",job="integrations/kubernetes/cadvisor", container="neo4j"}` I get the correct nodes:
![Screenshot from 2022-12-14 13-41-28](https://user-images.githubusercontent.com/13048718/207611686-2460c27e-05a3-41dc-82a1-aa04ab3125ea.png)

The expanded recorded query is:
```
container_memory_working_set_bytes{image!="",job="integrations/kubernetes/cadvisor", container="neo4j"} * on (namespace, pod) group_left (node) topk by (namespace, pod) (1, max by (namespace, pod, node) (kube_pod_info{node!=""}))
```

I'm not very familiar with this kind of queries, but I guess the aggregation is not correct in the case of having multiple clusters :thinking:

**Comment by AndresCalimero** [comment]
I spent some time trying to understand and fix the query, I came up with this query that includes the label `cluster` in the matching labels of the join:

```
container_memory_working_set_bytes{image!="",job="integrations/kubernetes/cadvisor", container="neo4j"} * on (namespace, pod, cluster) group_left (node) topk by (namespace, pod, cluster) (1, max by (namespace, pod, node, cluster) (kube_pod_info))
```

I think it should work even if the `cluster` label is not present, I don't know if it makes sense to assume that `cluster` is a wildly used label and therefore can be used in junction with the `namespace` and `pod` labels to uniquely identify a pod in a multi-cluster setup. I also have the feeling that `node_namespace_pod_container:container_memory_working_set_bytes` is not the only query that has issues when having a multi-cluster setup.

Let me know if I should create a PR with the updated query.
